### PR TITLE
Add the sphinx-togglebutton extension and a figure for MAP_FRAME_TYPES

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

This PR adds the [`sphinx-togglebutton`](https://sphinx-togglebutton.readthedocs.io/en/latest/) extension (~20 KB) to the repository, and also an example figure for MAP_FRAME_TYPE showing how the togglebutton works.

**Preview link:** https://gmt-p0akp8sm5-gmt.vercel.app/gmt.conf.html#term-MAP_FRAME_TYPE

It shows a toggle button on the right:

image

Click the button, and you will see:

image

Fixes #

**Reminders**

- [ ] Make sure that your code follows our style. Use the other functions/files as a basis.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Describe changes to function behavior and arguments in a comment below the function declaration.
- [ ] If adding new functionality, add a detailed description to the documentation and/or an example.

### CODE REVIEW

**Comment by meghanrjones** [comment]
Is the toggle button better than the dropdown directive for sphinx-panels?

**Comment by seisman** [comment]
> Is the toggle button better than the dropdown directive for sphinx-panels?

I just updated the PR description. 

`sphinx-togglebutton` provides two different styles for the toggle button (right figure). The first style (top one in the right figure) is very similar to the dropdown provided by `sphinx-panels` (left figure). Personally, I prefer the second style (bottom one in the right figure), because it's less distracted. 

| sphinx-panels | sphinx-togglebutton |
|---|---|
| image | image |

**Review by meghanrjones — APPROVED** [review]
Nice!

**Comment by seisman** [comment]
@meghanrjones Do you still want to try the `dropdown` directive as you mentioned in #4743

File: doc/rst/source/gmt.conf.rst [hunk 1]
```diff
@@ -677,6 +677,18 @@ MAP Parameters
...
+
+            Here is an example showing the apperance of different **MAP_FRAME_TYPE** settings.
+
+            .. literalinclude:: /_verbatim/GMT_map_frame_type.txt
+
+            .. figure:: /_images/GMT_map_frame_type.*
+               :width: 100%
+               :align: center
+
+               Apperance of different **MAP_FRAME_TYPE** settings
```
Comment (meghanrjones) [inline doc/rst/source/gmt.conf.rst:690]:
```suggestion
               Appearance of different **MAP_FRAME_TYPE** settings
```

File: doc/rst/source/gmt.conf.rst [hunk 2]
```diff
@@ -677,6 +677,18 @@ MAP Parameters
         head length and width are 10 and 5 times this width, respectively.  You
         may control its shape via :term:`MAP_VECTOR_SHAPE`.
 
+        .. toggle::
+
+            Here is an example showing the apperance of different **MAP_FRAME_TYPE** settings.
```
Comment (meghanrjones) [inline doc/rst/source/gmt.conf.rst:682]:
```suggestion
            Here is an example showing the appearance of different **MAP_FRAME_TYPE** settings.
```

**Comment by meghanrjones** [comment]
> @meghanrjones Do you still want to try the `dropdown` directive as you mentioned in #4743

Sure, I'll submit a PR. I think showing the code in a dropdown menu will be very helpful, but am not tied to the `dropdown` directive over the togglebutton extension. We can use the two PRs to compare - it will be best to be consistent throughout the documentation. Although we could also be consistent in using one option when both the figure and code are included in the dropbown and the other when the figure is static and the code is in the dropdown.

**Comment by meghanrjones** [comment]
Regarding [this comment](https://github.com/GenericMappingTools/gmt/pull/4927#issuecomment-792875334) and [this reply](https://github.com/GenericMappingTools/gmt/pull/4927#issuecomment-792882658) about the toggle button hint text, I agree that we'll want to use a least two different texts for different cases. Do you think it's better to implement that here or submit a feature request to sphinx-togglebutton for extending the directive to support a custom hint text?

**Comment by seisman** [comment]
> Regarding [this comment](https://github.com/GenericMappingTools/gmt/pull/4927#issuecomment-792875334) and [this reply](https://github.com/GenericMappingTools/gmt/pull/4927#issuecomment-792882658) about the toggle button hint text, I agree that we'll want to use a least two different texts for different cases. Do you think it's better to implement that in doc/rst/_extensions or submit a feature request to sphinx-togglebutton for extending the directive to support a custom hint text?

I think we can add it here first, and submit a feature request later. It's unclear to me if they like this feature and how often they made a release.

**Comment by meghanrjones** [comment]
> @meghanrjones I'd like to merge this PR into master first and work on the "custom hint text" in a separate PR. I feel that it's easier to track the changes and submit a PR to upstream.

Yes, I agree.

**Comment by seisman** [comment]
Just had a quick look at the sphinx-togglebutton source codes, it looks difficult to support "custom hint text", because the hint text is controlled by CSS.

I'll open a feature request later. Perhaps we can use "Click to show the source or an example" as a temporary workaround.

**Pull request merged**